libr/parse/p/parse_x86_pseudo.c

```diff
@@ -358,12 +364,58 @@ static inline void mk_reg_str(const char *regname, int delta, bool sign, bool at
 	r_strbuf_free (sb);
 }
 
-static bool varsub (RParse *p, RAnalFunction *f, ut64 addr, int oplen, char *data, char *str, int len) {
-	RList *bpargs, *spargs;
+static const RList *__args(RParse *p, RAnalFunction *f, int type) {
```

Comment:
add also a `r_return_val_if_fail` to check that type is either 's' or 'b'. Any other value should be refused.

Reply:
actually, instead of using 'b', 's', etc. you should use the defines `R_ANAL_VAR_KIND_BPV`, `R_ANAL_VAR_KIND_SPV`, etc.